BLU: Final Sting handling

Hugmeir opened the pull request:
`Final Sting` is one of BLU's "signature" moves --  At the cost of our lives, we do a significant chunk of damage.

The changes in this MR accomplish two things:

1. Deaths caused by `Final Sting` should not be counted as Deaths at all
2. `Final Sting` is normally buffed by using Moon Flute first; thus buff windows that end in a Final Sting are excempt from most requirements.

Also, note that all of this also applies to `Self-Destruct`.

Hugmeir committed:
BLU: Dying to Final Sting/Self-Destruct won't be counted as a death

Final Sting is essentially a pocket LB2; most BLU fights end with
most of the team exploding themselves for a huge burst of damage.

Hugmeir committed:
BLU: Moon Flute windows with Final Stings have no requirements

You whistle, tingle, Moon Flute, then do whatever you can and
pop Final Sting.  "Whatever you can" might be nothing at all;
and depending on the window, you might even Tingle in it.

Hugmeir committed:
Core Death module: allow overriding the death suggestion "why" field

Hugmeir committed:
BLU: Let people know that their Final Stings are not counted as Deaths

Hugmeir committed:
BLU: yarn extract for the new Final Sting-related messages

Hugmeir committed:
BLU Moon Flute + Final Sting: tsc & linter fixes

Hugmeir commented:
A sample log: https://www.fflogs.com/reports/yGdD6tr842KYMxVF#fight=2&source=8

AkairyuGestalter commented on src/parser/jobs/blu/modules/Death.tsx:
```diff
@@ -0,0 +1,68 @@
+				value={playerInfo.count}
+				_1="# death"
+				other="# deaths"
+			/>.  That number does not include the <Plural id="blu.deaths.why.final_stings"
```
This is really wordy. Can we do something like '<Plural [number of deaths] />, excluding those from Final Sting'? That should be enough to get the point across.

Hugmeir replied:
Yeah, that wording sucks. Hm... I was trying to avoid "excluding X" because it makes it a bit ambigious for the reader -- is that telling me that the deaths are not being counted? Or that I should not count one of the deaths there because it came from Final Sting?

Latest commit has different wording, going from this:

    2 deaths. That number does not include the 1 death from using Final Sting

To this:

    2 deaths, not counting 1 death from using Final Sting.

AkairyuGestalter commented on src/parser/jobs/blu/modules/MoonFlute.tsx:
```diff
@@ -132,7 +138,27 @@ export class MoonFlute extends BuffWindow {
+	}
+
+	private adjustExpectedActionOutcome(window: HistoryEntry<EvaluatedAction[]>, _action: TrackedActionGroup) {
+		const finalStingUsed = window.data.filter(event => (event.action.id === this.data.actions.FINAL_STING.id || event.action.id === this.data.actions.SELF_DESTRUCT.id)).length
```
Minor nit: Since you're running the same logic in both adjust functions, you might consider extracting that to a helper function

Hugmeir replied:
Done!

AkairyuGestalter reviewed (changes requested):
Overall this looks fine, just two smaller requests. Also looks to be worth a pull from master and a re-run of yarn extract to get rid of the merge conflicts on the i18n files.

Hugmeir committed:
BLU: Reword the Death suggestion we give if they used Final Sting

From this:

    2 deaths. That number does not include the 1 death from using Final Sting

To this:

    2 deaths, not counting 1 death from using Final Sting.

This commit also patches a minor bug -- We weren't supposed to show
the Final Sting message if they didn't use Final Sting, but the code
was missing a `return` so it was being shown regardless.

Hugmeir committed:
BLU Moon Flute + Final Sting: Extract some common functionality into a helper sub
